docs/queries/filters.md

```diff
@@ -125,29 +125,42 @@ because the tasks starts before tomorrow. Only one of the dates needs to match.
 
 ### Recurrence
 
--   `is recurring`
--   `is not recurring`
+- `is recurring`
+- `is not recurring`
 
 ### File Path
 
--   `path (includes|does not include) <path>`
-    -   Matches case-insensitive (disregards capitalization).
+- `path (includes|does not include) <path>`
+  - Matches case-insensitive (disregards capitalization).
 
 ### Description
 
--   `description (includes|does not include) <string>`
-    -   Matches case-insensitive (disregards capitalization).
-    -   Disregards the global filter when matching.
+- `description (includes|does not include) <string>`
+  - Matches case-insensitive (disregards capitalization).
+  - Disregards the global filter when matching.
 
 ### Heading
 
--   `heading (includes|does not include) <string>`
-    -   Whether or not the heading preceding the task includes the given string.
-    -   Always tries to match the closest heading above the task, regardless of heading level.
-    -   `does not include` will match a task that does not have a preceding heading in its file.
-    -   Matches case-insensitive (disregards capitalization).
+- `heading (includes|does not include) <string>`
+  - Whether or not the heading preceding the task includes the given string.
+  - Always tries to match the closest heading above the task, regardless of heading level.
+  - `does not include` will match a task that does not have a preceding heading in its file.
+  - Matches case-insensitive (disregards capitalization).
 
 ### Sub-Items
 
--   `exclude sub-items`
-    -   When this is set, the result list will only include tasks that are not indented in their file. It will only show tasks that are top level list items in their list.
+- `exclude sub-items`
+  - When this is set, the result list will only include tasks that are not indented in their file. It will only show tasks that are top level list items in their list.
+
+### Tags
+
+- `tags (include|do not include) <tag>`
```

Comment:
I'll add the other format as well. 


I was looking at query and was thinking about a refactor. Stripping down query to the logic to manage a set of filter objects and then each query type(line) being a separate class. 
This would cleanup the complexity of the class and move all the separate query lines to separate classes to be constructed and added to the 'query engine'. Query would then run thought the collection asking each filter if it matches and if it does return the filter. 

The collection could be loaded with the common ones first and break out when hit to speed up as much as possible. 

Just a idea I am playing with.

Reply:
Yes, I've been having similar thoughts. I especially want to have 

I like what the CONTRIBUTING page has to say about optionally chatting about things before implementing them.
https://github.com/schemar/obsidian-tasks/blob/main/CONTRIBUTING.md#updating-code

It sounds to me like a prime candidate to have a discussion beforehand in a new issue, to pool ideas.

I'm not saying you would, but please don't do any major refactoring right now on this issue, as it will conflict with my open PR and make that harder to merge.

Reply:
No, not refactoring. It was a idea to be discussed to make it more maintainable and easier for others to add more query types. 

I have updated the grammar types to include both.